daemon/delete.go

```diff
@@ -77,6 +77,22 @@ func (daemon *Daemon) rmLink(container *container.Container, name string) error
 // cleanupContainer unregisters a container from the daemon, stops stats
 // collection and cleanly removes contents and metadata from the filesystem.
 func (daemon *Daemon) cleanupContainer(container *container.Container, forceRemove, removeVolume bool) (err error) {
+	// If force removal is required, delete container from various
+	// indexes even if removal failed.
+	defer func() {
+		if err == nil || forceRemove {
```

Comment:
ok - can you rename inner-scope `err`s for clarifying the scopes?
People (including me 😅) can get confused when there are variables with same name but different scopes